tests/test_client.py

```diff
@@ -147,3 +153,225 @@ def on_message(client, userdata, msg):
 
         packet_in = fake_broker.receive_packet(1)
         assert not packet_in  # Check connection is closed
+
+
```

Comment:
One problem with this approach is that the tests may fail even if it's nothing to do with WebsocketWrapper because you're relying on all of the client logic and the Python networking stack. These are integration tests rather than unit tests. The style of the other tests in Test_client is since they are a direct migration of the old tests, which were written in this way.

The way I see it, there are 3 groups of things that need to be tested:

1. `ws_set_options` stores the expected values in the client object and provides them to the WebsocketWrapper
2. WebsocketWrapper provides a socket-like interface to the client
3. WebsocketWrapper sends and receives the correct data with the socket it is wrapping

By using test doubles, you can avoid any network layer logic at all, with the possible exception of the first group. Also, most of these are entirely independent of MQTT.